src/mito2/src/region_write_ctx.rs

```diff
@@ -167,6 +167,7 @@ impl RegionWriteCtx {
             &self.wal_entry,
             &self.wal_options,
         )?;
+        self.next_entry_id += 1;
```

Comment:
Why do we need to incr this field? I think it's of no use in kafka wal.

Reply:
The current codebase ensures that a write on a region will only produce one log entry. So the increment is simply override by a subsequent `set_next_entry_id` when writing to wal finishes. 
However, such an assumption may not be hold in the future. At that moment, a write on a region may produce multiple log entries. If their entry ids are the same, then we cannot reconstruct an entry from records it split into. Since the `entry_records` hashmap is indexed by `entry_id`.